### ISSUE TITLE
[$1000] bottom border line is missing for the logged-in user in Manage members

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#13824]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. navigate to settings > workspaces
2. press new workspace. it will create a new workspace
3. go to Manage members > check Logged-in user in list

## Expected Result:
bottom border should visible under Logged-in user

## Actual Result:
bottom border is not visible under Logged-in user

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] iOS / native

**Version Number:** 1.2.43-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

<img src="https://user-images.githubusercontent.com/43996225/209562302-9372e71c-b28d-4eca-99c5-e9212ab199b0.png" width="40%" height="40%">

<img src="https://user-images.githubusercontent.com/43996225/209562397-1c7aad4c-6b84-4803-b015-84d7faa6f405.PNG" width="40%" height="40%">

https://user-images.githubusercontent.com/43996225/209562591-cff51d4e-fadb-4f4b-b10b-53616e9fcb48.MP4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @jayeshmangwani 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1671897956983559

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b06179dc7a0779d9</li>
        <li>Upwork Job ID: 1608008807755079680</li>
        <li>Last Price Increase: 2022-12-28</li>
    </ul>
</details>

**Comment by daraksha-dk** [comment]
# Proposal
Root cause: wrong logic is being used here when passing border bottom styles
Solution: use _.isEmpty(item.errors) instead of !item.errors here (https://github.com/Expensify/App/blob/f2a01fdb22c254e6e87dc32013ff9cd72c54e080/src/pages/workspace/WorkspaceMembersPage.js#L260)

**Comment by Pujan92** [comment]
**Proposal**

For the owner it seems `errors` property exists with empty object like `errors: {}`, updation to the conditions for bottom border will resolve the issue.

```diff
diff --git a/src/pages/workspace/WorkspaceMembersPage.js b/src/pages/workspace/WorkspaceMembersPage.js
index 92ac6559f..65e7d556c 100644
--- a/src/pages/workspace/WorkspaceMembersPage.js
+++ b/src/pages/workspace/WorkspaceMembersPage.js
@@ -257,7 +257,7 @@ class WorkspaceMembersPage extends React.Component {
             <OfflineWithFeedback errorRowStyles={[styles.peopleRowBorderBottom]} onClose={() => this.dismissError(item)} pending
Action={item.pendingAction} errors={item.errors}>
                 <Hoverable onHoverIn={() => this.willTooltipShowForLogin(item.login, true)} onHoverOut={() => this.setState({sho
wTooltipForLogin: ''})}>
                     <TouchableOpacity
-                        style={[styles.peopleRow, !item.errors && styles.peopleRowBorderBottom, !canBeRemoved && styles.cursorDi
sabled]}
+                        style={[styles.peopleRow, (_.isUndefined(item.errors) || _.isEmpty(item.errors)) && styles.peopleRowBorderBottom, !canBeRemoved && styles.cursorDisabled]}
                         onPress={() => this.toggleUser(item.login, item.pendingAction)}
                         activeOpacity={0.7}
                     >
```

**Comment by Pujan92** [comment]
**Proposal 2(IMO preferred)**

We can avoid setting the empty error object for the user while creating the new workspace.

```diff
diff --git a/src/libs/actions/Policy.js b/src/libs/actions/Policy.js
index 0f6784e7d..038d8e941 100644
--- a/src/libs/actions/Policy.js
+++ b/src/libs/actions/Policy.js
@@ -745,7 +745,6 @@ function createWorkspace(ownerEmail = '', makeMeAdmin = false, policyName = '',
             value: {
                 [sessionEmail]: {
                     role: CONST.POLICY.ROLE.ADMIN,
-                    errors: {},
                 },
             },
         },
```

**Comment by tgolen** [comment]
🟢 for proposal by @daraksha-dk

**Comment by Pujan92** [comment]
@tgolen Isn't [this](https://github.com/Expensify/App/issues/13824#issuecomment-1366469860) makes more sense(as it causes the issue and that property not needed for success onyx object)?

**Comment by daraksha-dk** [comment]
@tgolen - Should I raise the PR or wait for C+ review as well

**Comment by Pujan92** [comment]
> Just sharing the screenshot of how that `error` property exist for loggedIn user only after creating new workspace. 

To add - it only occurs after creating the new workspace due to that unnecessary `error` property over there, on re-login it won't persist.

**Comment by tgolen** [comment]
@Pujan92 I don't think that code needs to be changed, no. I think that setting it to an empty object is fine and would ensure that any existing errors would always get cleared out of Onyx.

> @tgolen - Should I raise the PR or wait for C+ review as well

You can go ahead and raise the PR.

**Comment by Pujan92** [comment]
> @Pujan92 I don't think that code needs to be changed, no. I think that setting it to an empty object is fine and would ensure that any existing errors would always get cleared out of Onyx.

I am still not sure how it will help to clear as we are only making it on create workspace and it definitely creates a new object which I believe won't connect/update with the previous object.

I still prefer solution 2 but if it can not be considered anyway then I would like to argue for my [1st](https://github.com/Expensify/App/issues/13824#issuecomment-1366444921) proposal.
Logically we need 2 conditions to apply
1. isUndefined
2. isEmpty

isEmpty alone will work but this method is not intended to check for undefined values. It should be used when a property exists in the object. https://lodash.com/docs/4.17.15#isEmpty

Hope that is something we can consider for the better approach.

**Comment by Pujan92** [comment]
Just to add we are using a SET onyxMethod not merge over there. So to be honest my thought is when we found the RC then why not fix that and go with the workaround.

https://github.com/Expensify/App/blob/f2f68908d84396962f0774e8ceb448b08ea8309a/src/libs/actions/Policy.js#L743

**Comment by tgolen** [comment]
@Pujan92 I missed that it was using `Onyx.set()`. I think that's a mistake. It was changed from `merge` to `set` in https://github.com/Expensify/App/pull/11130 and I have a call with @jasperhuangg later today and I will ask him about this. `merge()` should always be preferred over `set()` for this. The guidance for using one vs the other is here: https://github.com/Expensify/react-native-onyx#should-i-use-merge-or-set-or-both

Historically, we've tried to make the view layer of the app as flexible as possible when dealing with Onyx data since there will always be issues with different Onyx schemas and different app versions. We haven't completely solved that yet, so while I could be OK with removing `errors: {}` if it truly doesn't need to be there, we still have to change the view layer regardless.

I also don't think it's necessary to use `_.isUndefined()`. We use underscore, not lodash, and `_.isEmpty(undefined)` works just fine.

**Comment by Pujan92** [comment]
> @Pujan92 I missed that it was using `Onyx.set()`. I think that's a mistake. It was changed from `merge` to `set` in #11130 and I have a call with @jasperhuangg later today and I will ask him about this. `merge()` should always be preferred over `set()` for this. The guidance for using one vs the other is here: https://github.com/Expensify/react-native-onyx#should-i-use-merge-or-set-or-both
> 
> Historically, we've tried to make the view layer of the app as flexible as possible when dealing with Onyx data since there will always be issues with different Onyx schemas and different app versions. We haven't completely solved that yet, so while I could be OK with removing `errors: {}` if it truly doesn't need to be there, we still have to change the view layer regardless.
> 
> I also don't think it's necessary to use `_.isUndefined()`. We use underscore, not lodash, and `_.isEmpty(undefined)` works just fine.

@tgolen For the Onyx I am fine with your comment but I just checked the underscore and the definition for isEmpty same like the loadash as the name define. I agree it will work for undefined but right approach will be to use isUndefined which I strongly believe .

Sharing the def for ref
Returns true if collection has no elements. For strings and array-like objects _.isEmpty checks if the length property is 0. For other objects, it returns true if the object has no enumerable own-properties. Note that primitive numbers, booleans and symbols are always empty by this definition.

**Comment by tgolen** [comment]
> do you think for this particular case are we going to use merge bcoz it's a creation operation and SET is the proper fit

Yes, you can see in the guidance that I linked to says as the first bullet "Use merge() when creating a new object".

Regarding `_.isEmpty()` We always prefer the most simple solution, not the most complex. Therefore if `_.isEmpty(undefined)` works, then that is the most simple and it is sufficient.

**Comment by Pujan92** [comment]
Ok, Thanks. I made my points already. :)

**Comment by Pujan92** [comment]
Sorry but wanted to make last comment, I am arguing much bcoz I thought better approach should be preferred over what works.

**Comment by bfitzexpensify** [comment]
Looks like https://github.com/Expensify/App/pull/13885 was deployed to production last week, expecting the automated next steps to populate tomorrow or the day after

**Comment by tgolen** [comment]
Ah, sorry, I tried to reapply the bug label to trigger the next steps. I'm not sure what we need to do to trigger them. I'll ask in Slack

**Comment by dylanexpensify** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] @bfitzexpensify A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the GH issue for creating the test here:
- [ ] @tgolen @sobitneupane The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/12301
- [ ] @tgolen @sobitneupane The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A - @Beamanator the author is aware and part of the discussion when this regression was caught. 
- [ ] @tgolen @sobitneupane A discussion in #contributor-plus has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] @bfitzexpensify Payment has been made to the issue reporter (if applicable)
- [ ] @bfitzexpensify Payment has been made to the contributor that fixed the issue (if applicable)
- [ ] @bfitzexpensify Payment has been made to the contributor+ that helped on the issue (if applicable) - N/A we merged this one immediately as a breaking change before becoming a deploy blocker.

**Comment by tgolen** [comment]
@sobitneupane Would you mind filling those out, please?

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

> * [ ]  @tgolen @sobitneupane The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/10556 

> * [ ]  @tgolen @sobitneupane The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/10556#issuecomment-1402047328

**Comment by tgolen** [comment]
@bfitzexpensify, we've checked off all our items on this one. Looks like there is just one final thing for you.

**Comment by bfitzexpensify** [comment]
Thanks for the bump @tgolen - we're all done here!
